src/components/GamePage.vue

```diff
@@ -121,18 +123,63 @@ keys["Alt"] = "Alt";
 keys["Meta"] = "Meta";
 keys[" "] = "space";
 
-//キーボード押したらひかる
-//ロジックとしては押したボタンのkey.valueを取得する。"."+key.valueというクラスリストを持つノードを探す。（それぞれに対応するクラスの名前はあらかじめひつ一つのclasslistに書いてある。そのノードに光る要素を加える。keyupしたらremove。
+
+let index = 0;
+
 const KeyDown = () => {
+  if(code.finishcode.length === index){
+    console.log("finished");
+    return;
+  }
+  console.log(KeyboardEvent.keyCode)
+  //1: 最初にスペースを押したら始まる
+  if(code.correctcode === "" && event.key === " "){
+    for(let i = 0; i < code.fullcode.length; i++){
+        if(code.fullcode[i] === "<"){
+          code.correctcode = code.fullcode.substring(0,i)
+          code.pointercode = code.fullcode.substring(i,i+1);
+          code.notyetcode = code.fullcode.substring(i+1);
+          index = i;
+          break;
+      }
+    }
+  }
+  //2: ポインターと打ったキーが同じかどうかの判定
+  else if(event.key === code.pointercode){
+    //2-1: もしも">"だったら次の"<"までインデックスをスキップ
+    if(event.key === ">"){
+      code.correctcode = code.fullcode.substring(0,index);
+      //今のindex以降で次にある"<"
+      for(let i = index; i < code.fullcode.length; i++){
+        if(code.fullcode[i] === "<"){
+          code.correctcode = code.fullcode.substring(0,i)
+          code.pointercode = code.fullcode.substring(i,i+1);
+          code.notyetcode = code.fullcode.substring(i+1);
```

Comment:
piniaで定義している状態は、コンポーネント側で状態の変更、更新をするのではなく、pinia側でその処理をするのがよいと思いました。
```suggestion
          setCorrectcode(getCodeIndexTo0);
          setPointerCode(getCodeIndexPlusOne);
          setNotyetCode(getCodeEndtoIndexPlusOne);
```
変数名が冗長なのは仮なので気にしないでください。
これは仮にpinia側でset,getメソッドを定義していた場合は、このような記述ができると思います。